x-pack/test/functional/services/ml/job_wizard_geo.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export function MachineLearningJobWizardGeoProvider({ getService }: FtrProviderContext) {
+  const comboBox = getService('comboBox');
+  const testSubjects = getService('testSubjects');
+
+  return {
+    async assertGeoFieldInputExists() {
+      await testSubjects.existOrFail('mlGeoFieldNameSelect > comboBoxInput');
+    },
+
+    async assertGeoFieldSelection(expectedIdentifier: string[]) {
+      const comboBoxSelectedOptions = await comboBox.getComboBoxSelectedOptions(
+        'mlGeoFieldNameSelect > comboBoxInput'
+      );
+      expect(comboBoxSelectedOptions).to.eql(
+        expectedIdentifier,
+        `Expected population field selection to be '${expectedIdentifier}' (got '${comboBoxSelectedOptions}')`
+      );
+    },
+
+    async selectGeoField(identifier: string) {
+      await comboBox.set('mlGeoFieldNameSelect > comboBoxInput', identifier);
+      await this.assertGeoFieldSelection([identifier]);
+    },
+
+    async assertSplitFieldInputExists() {
+      await testSubjects.existOrFail('mlMultiMetricSplitFieldSelect > comboBoxInput');
+    },
+
+    async assertSplitFieldSelection(expectedIdentifier: string[]) {
+      const comboBoxSelectedOptions = await comboBox.getComboBoxSelectedOptions(
+        'mlMultiMetricSplitFieldSelect > comboBoxInput'
+      );
+      expect(comboBoxSelectedOptions).to.eql(
+        expectedIdentifier,
+        `Expected split field selection to be '${expectedIdentifier}' (got '${comboBoxSelectedOptions}')`
+      );
+    },
+
+    async selectSplitField(identifier: string) {
+      await comboBox.set('mlMultiMetricSplitFieldSelect > comboBoxInput', identifier);
```

Comment:
Is `mlMultiMetricSplitFieldSelect` the correct data test subject here? Should the component's test subject be updated to match with the Geo wizard?

Reply:
@alvarezmelissa87 I see how you updated the service methods in 12cfc23 to use `mlSplitFieldSelect` instead of `mlMultiMetricSplitFieldSelect`. But AFAICS, the actually used entry is still `mlMultiMetricSplitFieldSelect` which makes the tests fail. We probably need to also update the [testSubject of the component](https://github.com/elastic/kibana/blob/main/x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/split_field/split_field.tsx#L65).

Reply:
Fixed in 334b359ed606846e8069b1e70bd2e997a39bd5b3